app/scripts/metamask-controller.js

```diff
@@ -2668,6 +2676,33 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Sets the Ledger Live preference to use for hardware wallet support
+   * @param {bool} val - the value representing if they want Ledger Live support
+   * @param {Function} cb - A callback function called when complete.
+   */
+  setLedgerLivePreference(val, cb) {
+    try {
+      this.preferencesController.setLedgerLivePreference(val);
+
+      // TODO: Calling async functionality in a sync function isn't great
```

Comment:
Is there any blocker to this being ab async function then? It seems like it should be.

Reply:
Good shout!